tree/30_generic_methods/user_shell.cf

```diff
@@ -0,0 +1,61 @@
+#####################################################################################
+# Copyright 2017 Normation SAS
+#####################################################################################
+#
+# This program is free software: you can redistribute it and/or modify
+# it under the terms of the GNU General Public License as published by
+# the Free Software Foundation, Version 3.
+#
+# This program is distributed in the hope that it will be useful,
+# but WITHOUT ANY WARRANTY; without even the implied warranty of
+# MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+# GNU General Public License for more details.
+#
+# You should have received a copy of the GNU General Public License
+# along with this program.  If not, see <http://www.gnu.org/licenses/>.
+#
+#####################################################################################
+
+# @name User shell
+# @description Define the shell of the user. User must already exist.
+# @documentation This method does not create the user.
+#
+# @parameter login User's login
+# @parameter shell User's shell
+# 
+# @class_prefix user_shell
+# @class_parameter login
+
+bundle agent user_shell(login, shell)
+{
+  vars:
+      "old_class_prefix"  string => canonify("user_shell_${login}");
+      "promisers"          slist => { @{this.callers_promisers}, cf_null }, policy => "ifdefined";
+      "class_prefix"      string => canonify(join("_", "promisers"));
+      "args"               slist => { "${login}", "${shell}" };
+
+  classes:
+      "${login}_exists" expression => userexists("${login}");
```

Comment:
"args" is used as parameter for logs and the class is used in another one. I chose to always use an unique class to control action execution for more clarity.